src/index.ts

```diff
@@ -42,6 +42,52 @@ export class DiskStream extends Readable {
 	}
 }
 
+class DiskTransformStream extends Transform {
+	// Adds the recorded writes to the original disk stream.
+	private position: number = 0;
+	private readonly chunks: Iterator<DiskChunk>;
+	private currentChunk: DiskChunk | null;
+
+	constructor(private readonly disk: Disk) {
+		super();
+		this.chunks = this.getKnownChunks();
+		this.currentChunk = this.chunks.next().value;
+	}
+
+	private *getKnownChunks(): Iterator<DiskChunk> {
+		for (const chunk of this.disk.knownChunks) {
+			if (chunk instanceof BufferDiskChunk) {
+				yield chunk;
+			}
+		}
+	}
+
+	public _transform(chunk: Buffer, enc: string, cb: () => void): void {
+		const start: number = this.position;
+		const end: number = start + chunk.length - 1;
+		const interval: Interval = [ start, end ];
+		while (this.currentChunk) {
+			if (intervalIntersection(this.currentChunk.interval(), interval)) {
+				const buf = this.currentChunk.data();
+				const startShift = this.currentChunk.start - start;
+				const endShift = this.currentChunk.end - end;
+				const sourceStart = -Math.min(startShift, 0);
```

Comment:
Why a negative `sourceStart` here – isn't that going to be OOB?

Reply:
@jhermsmeier 
`sourceStart` can not be negative here:

 * if `startShift` is negative, `sourceStart` will be `- startShift` : positive
 * if `startShift` is positive, `sourceStart` will be 0

This is equivalent to:
```typescript
const sourceStart = (startShift < 0) ? -startShift : 0
```

Reply:
I see; I think `const sourceStart = (startShift < 0) ? -startShift : 0` is a bit more clear to the reader

Reply:
That's correct, updated.